app/src/main/java/ch/epfl/favo/view/tabs/NearbyFavorList.java

```diff
@@ -0,0 +1,111 @@
+package ch.epfl.favo.view.tabs;
+
+import android.annotation.SuppressLint;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ListView;
+import android.widget.RadioButton;
+import android.widget.SearchView;
+import android.widget.TextView;
+
+import androidx.fragment.app.Fragment;
+import androidx.navigation.Navigation;
+
+import java.util.Objects;
+
+import ch.epfl.favo.MainActivity;
+import ch.epfl.favo.R;
+import ch.epfl.favo.favor.Favor;
+import ch.epfl.favo.util.CommonTools;
+import ch.epfl.favo.view.ViewController;
+import ch.epfl.favo.view.tabs.favorList.searchBarCoordinator;
+
+import static androidx.navigation.Navigation.findNavController;
+import static ch.epfl.favo.util.CommonTools.hideKeyboardFrom;
+
+/**
+ * View will contain list of favors requested in the past. The list will contain clickable items
+ * that will expand to give more information about them. This object is a simple {@link Fragment}
+ * subclass.
+ */
+public class NearbyFavorList extends Fragment{
+
+    private TextView tipTextView;
+    private ListView listView;
```

Comment:
Do we want to support manual updates of the list (e.g. with a scroll-down-to-refresh pattern)? We could use a SwipeRefreshLayout to wrap the list view and then simply add a listener for refreshing actions triggering the updateNearbyList() method in the MapsPage.

Reply:
i think its best to implement something like this: https://firebase.googleblog.com/2017/12/using-android-architecture-components.html

Reply:
That's a good idea, I look forward to integrate that in future